### PR TITLE
SQL log actually ensures its table exists

### PR DESCRIPTION
**Pull request by WhiteXZ**
Because it didn't before. Whoops.

### CODE REVIEW

File: TShockAPI/SqlLog.cs [hunk 1]
```diff
@@ -63,6 +64,20 @@ public SqlLog(IDbConnection db, string textlogFilepath, bool clearTextLog)
...
+			var table = new SqlTable("Logs",
+				new SqlColumn("ID", MySqlDbType.Int32) {AutoIncrement = true, Primary = true},
+				new SqlColumn("TimeStamp", MySqlDbType.Text),
+				new SqlColumn("LogLevel", MySqlDbType.Int32),
+				new SqlColumn("Caller", MySqlDbType.Text),
+				new SqlColumn("Message", MySqlDbType.Text)
+				);
+
+			var creator = new SqlTableCreator(db,
+											  db.GetSqlType() == SqlType.Sqlite
```
Comment (nicatronTg) [inline TShockAPI/SqlLog.cs]: Bad formatting?

Comment (WhiteXZ) [inline TShockAPI/SqlLog.cs]: Yeah not sure what happened there.

File: TShockAPI/SqlLog.cs [hunk 2]
```diff
@@ -232,9 +247,8 @@ public void Write(string message, TraceLevel level)
 					return;
 				}
 
-				_database.Query("INSERT INTO Logs (LogLevel, TimeStamp, Caller, Message) VALUES (@0, @1, @2, @3)",
-					level, DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture),
-					caller, message);
+				_database.Query("INSERT INTO Logs (TimeStamp, Caller, LogLevel, Message) VALUES (@0, @1, @2, @3)",
```
Comment (nicatronTg) [inline TShockAPI/SqlLog.cs]: By changing the table structure in this way, don't you break backwards compatibility with the already released MySQL log system?

File: TShockAPI/SqlLog.cs [hunk 3]
```diff
@@ -232,9 +247,8 @@ public void Write(string message, TraceLevel level)
...
+					DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture), caller, (int)level, message);
```
Comment (bladecoding) [inline TShockAPI/SqlLog.cs]: Should use DateTime.UtcNow

File: TShockAPI/SqlLog.cs (hunk 2 above)
Comment (WhiteXZ) [inline TShockAPI/SqlLog.cs]: There is no structural change, just a change to the INSERT statement?
(And seeing as the table was never created previously, there's no backwards compatibility to break)

File: TShockAPI/SqlLog.cs (hunk 3 above)
Comment (WhiteXZ) [inline TShockAPI/SqlLog.cs]: The text logger and the old TShock log uses and used DateTime.Now

Comment (nicatronTg) [inline TShockAPI/SqlLog.cs]: Time to get with the times and switch all of the code!

Comment (Olink) [inline TShockAPI/SqlLog.cs]: "I checked BanManager and it doesn't.  Get #REKT nerd." - bcccc aka "I'm high6 and I made SQL"

Comment (bladecoding) [inline TShockAPI/SqlLog.cs]: See: https://github.com/NyxStudios/TShock/blob/general-devel/TShockAPI/DB/BanManager.cs#L157

File: TShockAPI/SqlLog.cs (hunk 2 above)
Comment (nicatronTg) [inline TShockAPI/SqlLog.cs]: @WhiteXZ your previous statement would work if someone manually created a table (such as the way GrandPa-G did).

File: TShockAPI/TextLog.cs [hunk 4]
```diff
@@ -212,7 +212,7 @@ public void Write(string message, TraceLevel level)
 			}
 
 			var logEntry = string.Format("{0} - {1}: {2}: {3}",
-					DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture),
+					DateTime.UtcNow.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture),
```
Comment (nicatronTg) [inline TShockAPI/TextLog.cs]: Text log should probably not be in UTC, because it's going to cause issues when people's logs don't line up anymore.

Comment (nicatronTg) [inline TShockAPI/TextLog.cs]: My bad telling you to change this specific one.

**Comment by nicatronTg** [comment]
Looks good to me. cc @Olink

**Pull request merged**